contrib/database/sql/internal/mockdriver.go

```diff
@@ -0,0 +1,139 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package internal // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/sqltest"
+
+import (
+	"context"
+	"database/sql/driver"
+	"io"
+)
+
+// MockDriver implements a mock driver that captures and stores prepared and executed statements
+type MockDriver struct {
```

Comment:
Right, this could have so much more but I didn't want to try and imagine how it could be used in the future. Right now, this is used in a very simple single threaded context but I do think this could evolve when/if we have other use-cases that come up.

Reply:
Also, let me know if you have any comments on the naming. I did this a bit quickly to get the tests running but a few things came to mind as I was writing this:
* Those mocks are more like stubs. We had the mocktracer already here which is arguably not a true mock so I went with that name but I could see an argument for a different name. I wouldn't call it `Stub` but maybe `FakeDriver` or something like `TestDriver`. 
* What about `PreparedStmts` and `ExecutedQueries`. Would you prefer `Prepared` and `Executed`?

Reply:
I think this is all great. No need to complicate it. Perhaps `Prepared` and `Executed` is nicer in my book but I don't feel strongly 🙂

Reply:
I updated the names. My internal Gabriel voice cared more about this one than you did apparently.